packages/trie/src/trie/trie.ts

```diff
@@ -935,8 +932,8 @@ export class Trie {
    * Returns a copy of the underlying trie.
    * @param includeCheckpoints - If true and during a checkpoint, the copy will contain the checkpointing metadata and will use the same scratch as underlying db.
    */
-  copy(includeCheckpoints = true): Trie {
-    const trie = new Trie({
+  async copy(includeCheckpoints = true): Promise<Trie> {
```

Comment:
Note: if we keep this we need to mention in the docs (breaking).

Good cath though, we might want to re-visit the other `copy()` methods as well.

Reply:
i'll dig into it a little bit more, to see if it's actually necessary for it to be async.  

we encourage users to use the `await Trie.create()` method, but many of our tests are written with `new Trie()`.  So yes, either a bunch of rewrites to make it al async, or maybe i will find that the create method doesn't actually need to be async.

Reply:
No, the `create()` method needs to be async, that's the whole point why we introduced this method, to allow for async trie instantiation when using e.g. LevelDB as a database which has async reads and writes.

So it is rather our laziness that we still use (our own discouraged) way of using `new Trie()` directly and we should optimally rewrite the tests (goes for some other libraries as well, I guess at least VM and Blockchain? 🤔)